src/ensembl/src/shared/components/layout/StandardAppLayout.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { ReactNode } from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+
+import { ReactComponent as Chevron } from 'static/img/shared/chevron-right.svg';
+import { ReactComponent as CloseIcon } from 'static/img/shared/close.svg';
+
+import styles from './StandardAppLayout.scss';
+
+enum SidebarModeToggleAction {
+  OPEN = 'open',
+  CLOSE = 'close'
+}
+
+type SidebarModeToggleProps = {
+  onClick: () => void;
+  showAction: SidebarModeToggleAction;
+};
+
+type SidebarNavigationProps = {
+  links: Array<{
+    label: string;
+    isActive: boolean;
+  }>;
+  onChange: (index: number) => void;
+};
+
+type StandardAppLayoutProps = {
+  mainContent: ReactNode;
+  sidebarContent: ReactNode;
+  sidebarToolstripContent?: ReactNode;
+  sidebarNavigation: SidebarNavigationProps;
+  topbarContent: ReactNode;
+  isSidebarOpen: boolean;
+  onSidebarToggle: () => void;
+  isDrawerOpen: boolean;
+  drawerContent?: ReactNode;
+  onDrawerClose: () => void;
+};
+
+const StandardAppLayout = (props: StandardAppLayoutProps) => {
+  const mainClassnames = classNames(
+    styles.main,
+    { [styles.mainDefault]: props.isSidebarOpen },
+    { [styles.mainFullWidth]: !props.isSidebarOpen }
+  );
+
+  const sidebarWrapperClassnames = classNames(
+    styles.sideBarWrapper,
+    { [styles.sideBarWrapperOpen]: props.isSidebarOpen },
+    { [styles.sideBarWrapperClosed]: !props.isSidebarOpen },
+    {
+      [styles.sideBarWrapperDrawerOpen]:
+        'isDrawerOpen' in props ? props.isDrawerOpen : false
```

Comment:
Can't we instead use [nullish coalescing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing)? e.g. `props.isDrawerOpened ?? false`

Reply:
👍
Great idea. This looks as if it were specifically designed for typescript to prevent it complaining of a missing object field when you try to check this field directly 😄

Reply:
![image](https://user-images.githubusercontent.com/6834224/71245270-43112c00-230c-11ea-8432-476ce9c647f8.png)

Well, I'll just need to add a babel plugin for that :-)

Reply:
Yes, just like for optional chaining. Maybe we can add this with a new PR?

Reply:
Added the babel plugin in this PR.